scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/PipelineStatus.kt

```diff
@@ -104,4 +104,31 @@ open class PipelineStatus(val state: KafkaStreams.State?, var isError: Boolean)
 
     override var exception: Exception? = null
     override var message: String? = null
+
+    fun isActive(): Boolean {
+        return when {
+            this is StreamStarting || this is Started -> true
+            this.state in
+                setOf(
+                    KafkaStreams.State.CREATED,
+                    KafkaStreams.State.REBALANCING,
+                    KafkaStreams.State.RUNNING,
+                )
+            -> true
+            else -> false
+        }
+    }
+
+    fun isError(): Boolean {
+        return when {
+            this is Error -> true
```

Comment:
ditto

Reply:
The pipeline status is controlled both internally within the `Pipeline` object (for example when Kafka Streams transitions from one state to another), but can also be set externally (i.e if one determines via application logic external to the pipeline that an error has occured: for example, if we couldn't create all the Kafka topics required by the pipeline). 

`PipelineStatus.Error` is a way for objects external to the `Pipeline` to declare those kinds of errors. The other error states are controlled internally within the `Pipeline`.